bam_qc_metrics/bam_qc.py

```diff
@@ -17,6 +17,19 @@ class base:
     MAX_READ_LENGTH_KEY = 'max_read_length'
     UNMAPPED_READS_KEY = 'unmapped reads'
 
+    def read_package_version(self):
+        in_path = os.path.realpath(os.path.join(os.path.dirname(__file__), os.pardir, 'VERSION'))
```

Comment:
I think this will stop working if the class that subclasses `base` is moved to a different directory level (the code assumes a certain package structure).

I think the cleanest solution I have seen is in MultiQC. They use `pkg_resources.get_distribution("multiqc").version` (https://github.com/ewels/MultiQC/blob/master/multiqc/utils/config.py) to pull out the version number from the `setup.py` file. `pkg_resources` is part of `setuptools` so is part of the existing dependencies.

Reply:
On the ticket I linked to a discussion of how to single-source the version. There isn't any accepted best practice, I went with option (4) as the simplest: https://packaging.python.org/guides/single-sourcing-package-version/

The MultiQC approach is option (5).

But fair point about the base class, maybe (5) is better after all. I'll see how it looks.

Reply:
If you want to stick to option (4), it might be best to load the version in the root `__init__.py` file of `bam_qc_metrics`. That one should always stay one directory up from where `VERSION` lives.

Reply:
Yes, I think I'll do that or similar. I don't think (5) will work for our purposes, it seems to depend on having correct metadata on pypi.org.